service-base/src/main/java/org/eclipse/hono/service/registration/DeviceRegistryRestServerBase.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ * <p>
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * <p>
+ * Contributors:
+ * Bosch Software Innovations GmbH - initial creation
+ */
+
+package org.eclipse.hono.service.registration;
+
+import static java.net.HttpURLConnection.HTTP_OK;
+import static org.eclipse.hono.util.RegistrationConstants.ACTION_GET;
+
+import org.eclipse.hono.config.ServiceConfigProperties;
+import org.eclipse.hono.service.http.HttpServiceBase;
+import org.eclipse.hono.util.RegistrationConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * Base implementation for a REST interface which can be used to register devices.
+ *
+ * TODO: add all operations of the Registration API.
+ */
+public class DeviceRegistryRestServerBase extends HttpServiceBase<ServiceConfigProperties> {
```

Comment:
You think of something like the following?
`public class DeviceRegistryRestServerBase<T> extends HttpServiceBase<T extends ServiceConfigProperties> { 
...
}`